apps/projects/contracts/Projects.sol

```diff
@@ -287,6 +312,101 @@ contract Projects is AragonApp {
         emit FulfillmentAccepted(_repoId, _issueNumber, _bountyFulfillmentId);
     }
 
+    /**
+     * @notice apply to be assigned to this issue by submitting timeline and workplan
+     * @param _repoId the github repo id of the issue
+     * @param _issueNumber the github issue up for assignment
+     * @param _application IPFS hash for the applicant's proposed timeline and strategy
+     */
+    function requestAssignment(
+        bytes32 _repoId, 
+        uint256 _issueNumber, 
+        string _application
+    ) external isInitialized 
+    {
+        require(bytes(repos[_repoId].issues[_issueNumber].assignmentRequests[msg.sender]).length == 0, "User already applied for this issue");
+
+        repos[_repoId].issues[_issueNumber].applicants.push(msg.sender);
+        repos[_repoId].issues[_issueNumber].assignmentRequests[msg.sender] = _application;
```

Comment:
Do we want to do any validation to ensure the _application is an IPFS hash?

Reply:
Would this be to prevent people from creating garbage submissions outside of the frontend? Not sure it's necessary if this isn't a concern.

Reply:
I think it makes more sense to just check in in the FE, since string validation on-chain can be expensive